ccm_web/client/src/models/models.ts

```diff
@@ -31,3 +31,12 @@ export interface Globals {
   userLoginId: string
   course: Course
 }
+export interface SingleError {
+  message: string
+  failedInput: string
```

Comment:
WRT the problem described in #156…

_Posted by @pushyamig in https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/156#issue-945302924_
> The `failedInput` represents input that user provided, like for fetching the course name, it would be the course name.  Updating the course name it would be course new name. In case of creating new sections, it would be section name. The idea is that based on the request the respective feature API ( like getCourseName, putCourseName etc) will know how to deal with the errors.

Several of the API functions currently take one piece of data in its input payload.  Having this `failedInput` property contain a string that represents the input would probably be sufficient for most of those cases.  (That is, if you ignore instances where a number was the input, which is then converted to a string when it appears in `failedInput`.)  However, as we implement more complex functions, like creating many sections in one request or enrolling many users in one or more sections, it doesn't make sense to convert all that input to a string and put it in this single property.

Instead of a `failedInput` _string_ property, this needs to use `method` _string_, `request` _string_, and `payload` _object_ (or _"any"_-type, if that concept is supported) properties.

It could be used in the following cases…

* get course name error (getting course sections would be similar)
  `"method": "GET", "request": "/api/course/374350/name"`
* set course name error
  `"method": "PUT", "request": "/api/course/374350/name", "payload": {"newName": "test course for lsloan"}`
* create new sections error (I don't know the format of this request well, so this is approximate)
  `"method": "POST", "request": "/api/course/374350/sections", "payload": {"sections": ["sectionA", "sectionB"]}`

That way we'll know all of the parts of the request made that resulted in the error.

Reply:
@lsloan For Multiple create sections look like as below. For concurrent sections creation we grab the info that is needs to send to the user, Like what section failed and what is the reason for failure. I am not sure what additional value by adding `method` and `url`. I am not saying No, I am trying to understand.
```
"errors": [
    {
      "failedInput": "sec1",
      "message": "Error(s) from Canvas:  Unprocessable Entity"
    },
    {
      "failedInput": "sec2",
      "message": "Error(s) from Canvas:  Unprocessable Entity"
    },
    
  ]
```

Reply:
I'm suggesting that it would be helpful to have all of the information about the request included in the error message.  That will be helpful for debugging problems with the API.

Reply:
@lsloan based on the comments [here](https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/157/files#r670803544) the error response object is going to be an evolving thing. I think we can discuss further what would be useful. 

Currently the rational used what is useful info the FE needs to the user and based on that the object is framed.

Reply:
@pushyamig, I don't have anything else to add to the discussion other than what I wrote above.

The information I recommended to be included in the error data are what I would consider a minimum requirement.  This error data could also be logged for debugging and support purposes and would include almost everything that needs to be known to reproduce the request.

You mentioned something about over engineering recently.  I don't know whether it was about this suggestion.  I agree that can be a problem, but under engineering is also a problem.